src/i18nConfig.ts

```diff
@@ -4,6 +4,25 @@ export const LOCALE_FR_KEY = 'fr'
 
 export type Locale = 'fr' | 'en' | 'es'
 
+interface Request {
+  cookies: {
+    get: (name: string) => { value: string } | undefined
+  }
+  headers: {
+    get: (name: string) => string | null
+  }
+  nextUrl: {
+    searchParams: {
+      get: (name: string) => string | null
+    }
+  }
+}
+
+interface Config {
+  locales: readonly string[]
+  defaultLocale: string
+}
+
 const i18nConfig = {
```

Comment:
Don't we have an I18nConfig type ? It would avoid typing request and config manually